drivers/serial/Kconfig.ns16550

```diff
@@ -28,12 +28,6 @@ config UART_NS16550_DRV_CMD
 
 	  Says n if not sure.
 
-choice
```

Comment:
Removing choice is not a proper fix since these configs need to be mutually exclusive.

My suggestion would be to rename `UART_NS16{7,9}50` to `UART_NS16550_VARIANT_16{7,9}50` and add `UART_NS16550_VARIANT_16550` inside the choice as the default value.